console-framework-client/src/main/java/io/axoniq/console/framework/messaging/SpanMatcher.kt

```diff
@@ -0,0 +1,94 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.axoniq.console.framework.messaging
+
+import org.axonframework.util.MavenArtifactVersionResolver
+import java.util.*
+import java.util.function.Predicate
+
+/**
+ * Enum used by the {@link AxoniqConsoleSpanFactory} to determine if it should call certain methods, bases on the span
+ * name. This makes it possible to support the way spans are implemented by default for version 4.6 to 4.8 as well as
+ * the version from 4.9. It also makes it possible to accommodate for custom span implementations.
+ */
+enum class SpanMatcher(val pre49Predicate: Predicate<String>, val from49Predicate: Predicate<String>) {
+    HANDLER(
+            Predicate { name: String ->
+                name == "QueryProcessingTask" ||
+                        name == "AxonServerCommandBus.handle" ||
+                        name == "DeadlineJob.execute"
+            },
+            Predicate { name: String ->
+                name == "QueryBus.processQueryMessage" ||
+                        name == "CommandBus.handleCommand" ||
+                        name.startsWith("DeadlineManager.executeDeadline(")
+            }),
+    OBTAIN_LOCK(
+            Predicate { name: String -> name == "LockingRepository.obtainLock" },
+            Predicate { name: String -> name == "Repository.obtainLock" }),
+    LOAD(
+            Predicate { name: String -> name.contains(".load ") },
+            Predicate { name: String -> name == "Repository.load" }),
+    COMMIT(
+            Predicate { name: String -> name.endsWith(".commit") },
+            Predicate { name: String -> name == "EventBus.commitEvents" }),
+    MESSAGE_START(
+            Predicate { name: String ->
+                name.endsWith("Bus.handle")
+                        || name == "SimpleQueryBus.query"
+                        || name.startsWith("SimpleQueryBus.scatterGather")
+                        || name.startsWith("PooledStreamingEventProcessor")
+                        || name.startsWith("TrackingEventProcessor")
+            },
+            Predicate { name: String ->
+                name == "CommandBus.dispatchCommand"
+                        || name == "QueryBus.query"
+                        || name.startsWith("QueryBus.scatterGatherQuery")
+                        || name.startsWith("StreamingEventProcessor")
+            })
+}
+
+class SpanMatcherPredicateMap : EnumMap<SpanMatcher, Predicate<String>>(SpanMatcher::class.java)
+
+private fun pre49PredicateMap(): SpanMatcherPredicateMap {
```

Comment:
Can we encapsulate this to an `object` class or a companion object inside another one? I'm not that fond of top-level functions if they are not widely used extensions

Reply:
Done, it's indeed cleaner.